src/Raven.Server/Documents/Sharding/Handlers/Processors/Streaming/ShardedStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,237 @@
+
+            //TODO stav: should have this?
+            //if(queryProcessor.IsIncludes())
+            //    throw new NotSupportedInShardingException("Includes is not supported in sharded streaming queries");
+
+            queryProcessor.Initialize(out Dictionary<int, BlittableJsonReaderObject> queryTemplates);
+
+            var cmds = new Dictionary<int, PostQueryStreamCommand>(RequestHandler.DatabaseContext.ShardCount);
+            foreach(var (shard, queryTemplate) in queryTemplates)
+            {
+                cmds.Add(shard, new PostQueryStreamCommand(queryTemplate, debug, ignoreLimit));
+            }
+
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Stav, DevelopmentHelper.Severity.Normal, "Handle continuation token in streaming");
+            //var continuationToken = RequestHandler.ContinuationTokens.GetOrCreateContinuationToken(context, query.Offset ?? 0, query.Limit ?? Int32.MaxValue);
+
+            IComparer<BlittableJsonReaderObject> comparer = query.Metadata.OrderBy?.Length > 0
+                ? new ShardedDocumentsComparer(query.Metadata, isMapReduce: false)
+                : new DocumentBlittableLastModifiedComparer();
+
+            var op = new ShardedStreamQueryOperation(HttpContext, () =>
+            {
+                IDisposable returnToContextPool = ContextPool.AllocateOperationContext(out JsonOperationContext ctx);
+                return (ctx, returnToContextPool);
+            }, comparer, cmds,  skip: query.Offset ?? 0, take: query.Limit ?? Int32.MaxValue, token.Token);
+
+            return await RequestHandler.ShardExecutor.ExecuteParallelForShardsAsync(cmds.Keys.ToArray() ,op, token.Token);
+        }
+
+        protected override async ValueTask ExecuteAndWriteQueryStreamAsync(TransactionOperationContext context, IndexQueryServerSide query, string format,
+            string[] propertiesArray, string fileNamePrefix, bool ignoreLimit, bool _, OperationCancelToken token)
+        {
+            //writer is blittable->document or blittable->csv
+
+            var (results, queryStatistics) = await ExecuteQueryAsync(context, query, null, ignoreLimit, token);
+            await using (var writer = GetBlittableQueryResultWriter(format, isDebug: false, context, HttpContext.Response, RequestHandler.ResponseBodyStream(), fromSharded: false,
+                             propertiesArray, fileNamePrefix))
+            {
+                var queryResult = new StreamDocumentIndexEntriesQueryResult(HttpContext.Response, writer, token);// writes blittable docs as blittable docs
+                queryResult.TotalResults = queryStatistics.TotalResults;
+                queryResult.IndexName = queryStatistics.IndexName;
+                queryResult.IndexTimestamp = queryStatistics.IndexTimestamp;
+                queryResult.IsStale = queryStatistics.IsStale;
+                //queryStatistics.ResultEtag
```

Comment:
todo? how we want to handle this? merge all of them?

Reply:
changed to use `Hashing.Combine(long, long)` as done in `ShardedQueryProcessor`